app/lib/handlers.dart

```diff
@@ -450,6 +457,28 @@ Future<shelf.Response> apiPackagesHandler(shelf.Request request) async {
   return _jsonResponse(json);
 }
 
+/// Handles requests for /flutter
+Future<shelf.Response> flutterHandler(shelf.Request request) async {
+  final String path = request.requestedUri.path;
+  final String basePath = '/flutter/plugins';
+  if (path == '/flutter' || path == '/flutter/') {
+    return _redirectResponse(basePath);
+  }
+  final int page = _pageFromUrl(request.url);
+  if (path.endsWith('.json')) {
+    return packagesHandlerJson(request, page, true,
+        basePath: basePath, detectedType: BuiltinTypes.flutterPlugin);
+  } else if (request.url.queryParameters['format'] == 'json') {
+    return packagesHandlerJson(request, page, false,
+        basePath: basePath, detectedType: BuiltinTypes.flutterPlugin);
```

Comment:
Do we need to expose these two json api endpoints? Who is going to use them?

(These `.json` and `?format=json` apis are not the real apis used by the pub client. )